bookkeeper-server/src/main/java/org/apache/bookkeeper/tools/cli/commands/bookie/RegenerateInterleavedStorageIndexFileCommand.java

```diff
@@ -0,0 +1,123 @@
+import lombok.Setter;
+import lombok.experimental.Accessors;
+import org.apache.bookkeeper.bookie.InterleavedStorageRegenerateIndexOp;
+import org.apache.bookkeeper.conf.ServerConfiguration;
+import org.apache.bookkeeper.tools.cli.helpers.BookieCommand;
+import org.apache.bookkeeper.tools.framework.CliFlags;
+import org.apache.bookkeeper.tools.framework.CliSpec;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Command to regenerate an index file for interleaved storage.
+ */
+public class RegenerateInterleavedStorageIndexFileCommand
+    extends BookieCommand<RegenerateInterleavedStorageIndexFileCommand.RISIFFlags> {
+
+    static final Logger LOG = LoggerFactory.getLogger(RegenerateInterleavedStorageIndexFileCommand.class);
+
+    private static final String NAME = "regenerate-interleaved-storage-index-file";
+    private static final String DESC =
+        "Regenerate an interleaved storage index file, from available entrylogger " + "files.";
+    private static final String DEFAULT = "";
+
+    public RegenerateInterleavedStorageIndexFileCommand() {
+        this(new RISIFFlags());
+    }
+
+    private RegenerateInterleavedStorageIndexFileCommand(RISIFFlags flags) {
+        super(CliSpec.<RISIFFlags>newBuilder()
+                  .withName(NAME)
+                  .withDescription(DESC)
+                  .withFlags(flags)
+                  .build());
+    }
+
+    /**
+     * Flags for regenerate interleaved storage index file command.
+     */
+    @Accessors(fluent = true)
+    @Setter
+    public static class RISIFFlags extends CliFlags {
+
+        @Parameter(names = { "-p", "--password" },
+            description = "The bookie stores the password in the index file, so we need it to regenerate."
+                          + "This must match the value in the ledger metadata.")
+        private String password = DEFAULT;
+
+        @Parameter(names = { "-b", "--b64password" },
+            description = "The password in base64 encoding, for cases where the password is not UTF-8.")
+        private String b64Password = DEFAULT;
+
+        @Parameter(names = { "-d", "--dryrun" }, description = "Process the entryLogger, but don't write anthing.")
+        private boolean dryRun;
+
+        @Parameter(names = { "-l", "--ledgerids" },
+            description = "Ledger(s) whose index needs to be regenerated. Multiple can be specified, comma separated.",
+            splitter = CommaParameterSplitter.class)
+        private List<Long> ledgerIds;
+
+    }
+
+    @Override
+    public boolean apply(ServerConfiguration conf, RISIFFlags cmdFlags) {
+        try {
+            return generate(conf, cmdFlags);
+        } catch (Exception e) {
+            throw new UncheckedExecutionException(e.getMessage(), e);
+        }
+    }
+
+    private boolean generate(ServerConfiguration conf, RISIFFlags flags) throws NoSuchAlgorithmException, IOException {
+        byte[] password;
+        if (!flags.password.equals(DEFAULT)) {
```

Comment:
password can be `null`, no?

I think you also need to add a test to cover if password is null.